test/python/circuit/test_uniform_superposition_gate.py

```diff
@@ -0,0 +1,139 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Uniform Superposition Gate test.
+"""
+
+import unittest
+import math
+from test import QiskitTestCase
+import numpy as np
+from ddt import ddt, data
+
+from qiskit import QuantumCircuit
+from qiskit.quantum_info import Operator
+from qiskit.compiler import transpile
+
+from qiskit.circuit.library.data_preparation import (
+    UniformSuperpositionGate,
+)
+
+
+@ddt
+class TestUniformSuperposition(QiskitTestCase):
+    """Test initialization with UniformSuperpositionGate class"""
+
+    @data(2, 3, 5)
+    def test_uniform_superposition_gate(self, num_superpos_states):
+        """Test Uniform Superposition Gate"""
+        n = int(math.ceil(math.log2(num_superpos_states)))
+        desired_sv = (1 / np.sqrt(num_superpos_states)) * np.array(
+            [1.0] * num_superpos_states + [0.0] * (2**n - num_superpos_states)
+        )
+        gate = UniformSuperpositionGate(num_superpos_states, n)
+        qc = QuantumCircuit(n)
+        qc.append(gate, list(range(n)))
+        unitary_matrix = np.array(Operator(qc).data)
+        actual_sv = unitary_matrix[:, 0]
```

Comment:
You should be able to just use
```suggestion
        actual_sv = Statevector(gate)
```
where `Statevector` lives in `qiskit.quantum_info`

Reply:
Okay, I have used that. Thank you, Julien!